lib/arborist/build-ideal-tree.js

```diff
@@ -1421,7 +1421,7 @@ This is a one-time fix-up, please be patient...
     // prune anything deeper in the tree that can be replaced by this
     if (this.idealTree) {
       for (const node of this.idealTree.inventory.query('name', newDep.name)) {
-        if (node.isDescendantOf(target))
+        if (!node.isTop && node.isDescendantOf(target))
```

Comment:
this extra check is because it's possible for `isDescendantOf` to return true when the node `isTop`? when is that the case? i'm wondering if it's worth putting a comment here explaining why both tests are needed, because i would sure assume that a node couldn't be both the descendant of a target _and_ a top node